Dataset Creation UI fixes & Improvements

leiyre opened the pull request:
- [x] Rating values validate min / max
- [x] Question name format and renaming
- [x] Minor style adjustments 
- [x] Show required / not required in the form preview
- [x] Redirect from /new/repoId to /repoId
- [x] Redirect to error page when repoId is invalid

leiyre committed:
fix rating value min / max validation

leiyre committed:
format and replace question name on changing type

leiyre committed:
improve styles

leiyre committed:
show required in the form preview

leiyre committed:
redirect /new/repoId to /repoId

leiyre committed:
redirect to error page

frascuchon committed:
chore: Drop me

frascuchon committed:
Merge branch 'develop' into fix/dataset-creation-ui-improvements

frascuchon commented on argilla-frontend/v1/infrastructure/repositories/HubRepository.ts:
```diff
@@ -19,8 +23,11 @@ export class HubRepository {
-      return {};
+    } catch (e) {
+      throw {
+        status: e.response.status,
```
Can we map the status to 404 if the error status is 401? As a user is a bit weird to see the big 401 number on the error page. We can iterate later on this when supporting private datasets, where the 401 makes more sense.
<img width="1512" alt="Captura de pantalla 2024-11-06 a las 16 17 10" src="https://github.com/user-attachments/assets/cc60c87b-5b3a-46bd-a055-36f1d8064583">

leiyre replied:
Sure!

frascuchon replied:
Also, I think we should define a copy message instead of the `ERROR_FETCHING_DATASET`. What about "Cannot fetch the dataset"?

frascuchon committed:
[BUGFIX] `argilla frontend`: redirect after login (#5635)

# Description
<!-- Please include a summary of the changes and the related issue.
Please also include relevant motivation and context. List any
dependencies that are required for this change. -->

Closes https://github.com/argilla-io/argilla/issues/5633

**Type of change**
<!-- Please delete options that are not relevant. Remember to title the
PR according to the type of change -->

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

**How Has This Been Tested**
<!-- Please add some reference about how your feature has been tested.
-->

**Checklist**
<!-- Please go over the list and make sure you've taken everything into
account -->

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature
works
- I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

---------

Co-authored-by: José Francisco Calvo <jose@argilla.io>
Co-authored-by: José Francisco Calvo <josefranciscocalvo@gmail.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

frascuchon committed:
Merge branch 'develop' into fix/dataset-creation-ui-improvements

frascuchon commented on argilla-frontend/components/features/dataset-creation/configuration/DatasetConfigurationForm.vue:
```diff
@@ -139,21 +137,21 @@ export default {
     addQuestion(type) {
-      const questionName = `${type} ${this.dataset.selectedSubset.questions.length}`;
+      const name = this.$t(`config.questionId.${type}`);
+      const questionName = `${name}-${this.dataset.selectedSubset.questions.length}`;
```
I think the idea in the discussion was to use the `_` instead of `-`. Let's make the change if there are no technical limitations to that. 

```suggestion
      const questionName = `${name}_${this.dataset.selectedSubset.questions.length}`;
```

frascuchon committed:
Merge branch 'develop' into fix/dataset-creation-ui-improvements

leiyre committed:
fix question name generator

leiyre committed:
include validation for rating

leiyre committed:
show error 404

leiyre committed:
fix question name generator when there is no number

frascuchon committed:
Merge branch 'develop' into fix/dataset-creation-ui-improvements

leiyre committed:
show static error message

leiyre committed:
Merge branch 'fix/dataset-creation-ui-improvements' of https://github.com/argilla-io/argilla into fix/dataset-creation-ui-improvements

frascuchon committed:
Merge branch 'develop' into fix/dataset-creation-ui-improvements

leiyre committed:
fix generate max if question without number

frascuchon commented on argilla-frontend/components/features/dataset-creation/configuration/DatasetConfigurationForm.vue:
```diff
@@ -144,7 +145,7 @@ export default {
         ...this.dataset.selectedSubset.questions.map((question) => {
           const numberInName = question.name.split("_").pop();
-          return parseInt(numberInName);
+          return parseInt(numberInName) || 0;
```
How can a mapped column named "column_10" influence this?

leiyre replied:
In that case the next created question name will be "questionType_11"

leiyre committed:
remove data from template

leiyre committed:
update CHANGELOG
